### PR TITLE
ec2_eni - Migrate to boto3

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

ec2_eni: Move to boto3, add support for tags and name
    
Initial port from https://github.com/ansible/ansible/pull/54950
    Co-authored-by: @mike-healey
    Co-authored-by: @caphrim007
    Co-authored-by: @s-hertel 

fixes: #141 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ec2_eni

##### ADDITIONAL INFORMATION

### CODE REVIEW

**Comment by tremble** [comment]
Current state of play:
- Code 'generally' works
- There are some weird test flakes, I suspect some of the calls are either failing to return or are returning too soon:
"subnet_id is required when creating a new ENI"
https://app.shippable.com/github/ansible-collections/amazon.aws/runs/679/26/console
source_dest_check not updating properly - eventual consistency?
https://app.shippable.com/github/ansible-collections/amazon.aws/runs/679/24/console

File: plugins/modules/ec2_eni_info.py [hunk 1]
```diff
@@ -17,10 +17,17 @@
 author: "Rob White (@wimnat)"
 requirements: [ boto3 ]
 options:
+  eni_id:
+    description:
+      - The ID of the ENI.
+      - This option is mutually exclusive of I(filters).
+    type: str
+    version_added: 1.2.0
```
Comment (jillr) [inline plugins/modules/ec2_eni_info.py:25]:
```suggestion
    version_added: 1.3.0
```
Same for other new options

File: shippable.yml [hunk 2]
```diff
@@ -16,7 +16,8 @@ matrix:
     - env: T=units/3.6/1 A_REV=devel
     - env: T=units/3.7/1 A_REV=devel
     - env: T=units/3.8/1 A_REV=devel
-    - env: T=units/3.9/1 A_REV=devel
+#   Until boto/botocore#2002 is fixed
```
Comment (jillr) [inline shippable.yml:19]: We shouldn't need these commented out any longer

**Review by jillr — COMMENTED** [review]
Just a couple minor updates otherwise lg, thanks from bringing this over!

**Review by jillr — APPROVED** [review]

**Pull request merged**